pkg/receive/hashring.go

```diff
@@ -6,14 +6,31 @@ package receive
 import (
 	"context"
 	"fmt"
+	"sort"
+	"strconv"
 	"sync"
 
+	"github.com/cespare/xxhash"
+
 	"github.com/pkg/errors"
 	"github.com/thanos-io/thanos/pkg/store/labelpb"
 
 	"github.com/thanos-io/thanos/pkg/store/storepb/prompb"
 )
 
+// HashringAlgorithm is the algorithm used to distribute series in the ring.
+type HashringAlgorithm string
+
+const (
+	AlgorithmHashmod    HashringAlgorithm = "hashmod"
+	AlgorithmConsistent HashringAlgorithm = "consistent"
```

Comment:
Should this be called "ketama" instead of "consistent"? This that other consistent hashing algorithms can be added in the future without breaking backwards compatibility.

Reply:
I agree, we should give it a more specific name. We can change it to either `ring-hash`, `ketama`, `dynamo` or something along those lines. I'm open to suggestions :)

Reply:
`ketama` sounds like a good one to me. It's one algorithm that uses a ring-hash. There could be other algorithms using ring-hashes and then we would have a name conflict.

Reply:
Sounds good 👍 Renamed to `ketama`